app/assets/javascripts/components/sampleUpload.jsx

```diff
@@ -0,0 +1,336 @@
+
+  componentDidMount() {
+    this.initializeSelectTag();
+    $(ReactDOM.findDOMNode(this.refs.projectSelect)).on('change',this.handleProjectChange);
+    $(ReactDOM.findDOMNode(this.refs.hostSelect)).on('change',this.handleHostChange);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.clearError();
+    if(!this.isFormInvalid()) {
+      this.createSample()
+    }
+  }
+
+  initializeSelectTag() {
+    $('select').material_select();
+  }
+
+
+
+  clearError() {
+    this.setState({ 
+      invalid: false,
+      success: false
+     })
+  }
+
+  gotoPage(path) {
+    location.href = `${path}`
+  }
+
+  componentDidUpdate() {
+    this.handleProjectSubmit ? this.handleProjectSubmit : null
+  }
+
+  handleProjectSubmit(e) {
+    e.preventDefault();
+    this.clearError();
+    if(!this.isProjectInvalid()) {
+      this.addProject()
+    }
+  }
+
+  addProject() {
+    var that = this;
+    axios.post('/projects.json', {
+      project: {
+        name: this.refs.new_project.value,
+      },
+      authenticity_token: this.csrf
+    })
+    .then((response) => {
+      var newProjectList = that.state.allProjects.slice();
+      newProjectList.push(response.data);
+      that.setState({
+        allProjects: newProjectList,
+        project: response.data.name,
+        success: true,
+        successMessage: 'Project added successfully'
+      }, () => {
+        this.refs.new_project.value = '';
+        that.initializeSelectTag();
+      });
+    })
+    .catch((error) => {
+      that.setState({
+        invalid: true,
+        errorMessage: 'Project exists already or is invalid'
+      })
+    });
+  }
+
+  isProjectInvalid() {
+    if (this.refs.new_project.value === '') {
+      this.setState({
+        invalid: true,
+        errorMessage: 'Please enter valid project name'
+      })
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  createSample() {
+    var that = this;
+    axios.post('/samples.json', {
+      sample: {
+        name: this.refs.name.value.trim(),
+        project_name: this.state.project.trim(),
+        input_files_attributes: [{source_type: 's3', source: this.refs.first_file_source.value.trim() },
+        {source_type: 's3', source: this.refs.second_file_source.value.trim() }],
+        s3_preload_result_path: this.refs.s3_preload_result_path.value.trim(),
+        job_queue: this.state.job_queue,
+        memory: this.state.memory,
+        host_genome_id: this.state.hostId,
+        status: 'created'
+      },
+      authenticity_token: this.csrf
+    })
+    .then(function (response) {
+      that.setState({
+        success: true,
+        successMessage: 'Sample created successfully'
+      }, () => {
+        that.gotoPage(`/samples/${response.data.id}`);
+      })
+    })
+    .catch(function (error) {
+     that.setState({
+      invalid: true,
+       errorMessage: 'Failed to create sample'
+     })
+    });
+  }
+
+  filePathValid(str) {
+    var regexPrefix = /^s3:\/\//;
```

Comment:
can you trim the white space on both end of the input. It's a common mistake people make.